torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -360,26 +360,18 @@ def pre_load_state_dict_hook(module, state_dict, prefix, local_metadata, strict,
     _recurse_update_module(module, state_dict, prefix)
 
 def _recurse_update_module(module, state_dict, prefix):
-    for attr_name, attr in module.__dict__.items():
```

Comment:
nit: given that we don't need the recursion anymore, shall we remove these two functions and put the main logic in `state_dict_hook` and `pre_load_state_dict_hook`?

Reply:
Sounds good to me.  @pritamdamania87 do you agree?

Reply:
@awaelchli Yes this makes sense

Reply:
Done